Client/Frontend/Browser/User Scripts/UserScriptType.swift

```diff
@@ -8,64 +8,80 @@ import WebKit
 
 /// An enum representing a specific (modified) variation of a local script replacing any dynamic variables.
 enum UserScriptType: Hashable {
-    /// This type does farbling protection and is customized for the provided eTLD+1
-    /// Has a dependency on `nacl`
-    case farblingProtection(etld: String)
-    /// Scripts specific to certain domains
-    case domainUserScript(DomainUserScript)
-    /// A symple encryption library to be used by other scripts
-    case nacl
+  /// This type does farbling protection and is customized for the provided eTLD+1
+  /// Has a dependency on `nacl`
+  case farblingProtection(etld: String)
+  /// Scripts specific to certain domains
+  case domainUserScript(DomainUserScript)
+  /// A symple encryption library to be used by other scripts
+  case nacl
+  /// Enables de-amp ad-block rule.
+  ///
+  /// Works in conjunction with the injected de-amp script here: https://github.com/brave/adblock-resources/blob/master/resources/de-amp.js
+  ///
+  /// - Note: The de-amp script is injected using ad-block rules here:
+  ///  https://github.com/brave/adblock-lists/blob/master/brave-lists/brave-specific.txt
+  case deAMP
 
-    /// Return a source typ for this script type
-    var sourceType: ScriptSourceType {
-        switch self {
-        case .farblingProtection:
-            return .farblingProtection
-        case .domainUserScript(let domainUserScript):
-            switch domainUserScript {
-            case .youtubeAdBlock:
-                return .youtubeAdBlock
-            case .archive:
-                return .archive
-            case .braveSearchHelper:
-                return .braveSearchHelper
-            case .braveTalkHelper:
-                return .braveTalkHelper
-            }
-        case .nacl:
-            return .nacl
-        }
+  /// Return a source typ for this script type
+  var sourceType: ScriptSourceType {
+    switch self {
+    case .farblingProtection:
+      return .farblingProtection
+    case .deAMP:
+      return .deAMP
```

Comment:
Why are we injecting `deAmp` on `decidePolicyFor` instead of in `UserScriptManager` globally?